tests/dialects/test_x86.py

```diff
@@ -0,0 +1,7 @@
+from xdsl.dialects import x86
+
+
+def test_register():
+    rax = x86.register.RAX
+    assert rax.is_allocated
+    assert rax.register_name == "rax"
```

Comment:
Could you parametrize this to test all the registers? You can see an eample [here](https://github.com/xdslproject/xdsl/blob/main/tests/dialects/test_arith.py#L124)